### PR TITLE
Incorporated skeleton of MAP ingestion process

### PR DESCRIPTION
**Pull request by CBroz1**
- Added BehaviorIngest table and load function to mimic MAP functionality
- Provided examples for loading from matfile to Trial and Event tables

### CODE REVIEW

File: adamacs/behavior.py [hunk 1]
```diff
@@ -113,8 +93,50 @@ class TrialVariable(dj.Part):
...
+@schema
+class BehaviorIngest(dj.Imported):
+	definition = """
+	-> session.Recording
+	"""
+  @property
+  def key_source(self): #opional means of subsetting or reordering keys worth ingesting
+  	good_records = session.Recording.fetch('KEY') #& session.Qualitycontrol
+  	return good_records
+
```
Comment (ttngu207) [inline adamacs/behavior.py]:
```suggestion
```

Let's remove it here, modifying `key_source` is mostly not needed. We can add when needed

File: adamacs/behavior.py [hunk 2]
```diff
@@ -113,8 +93,50 @@ class TrialVariable(dj.Part):
...
+	def make(self, key): # reading bpod data to populate
+		# could model dir navigation after element_array_ephys
+		# which uses config file for root dir and csv for relative paths
+		# https://github.com/datajoint/workflow-array-ephys/blob/main/workflow_array_ephys/paths.py
+		bpod_root_dir = pathlib.Path(get_beh_root_dir(key))
+		bpod_sess_dir = pathlib.Path(get_beh_sess_dir(key))
+		bpod_dir = find_full_path(bpod_root_dir,bpod_sess_dir)
+
+		for file in os.listdir(bpod_dir): if filename.endswith(".mat"):
+			trial_info = load_bpod_matfile(key, bpod_dir + file)
```
Comment (ttngu207) [inline adamacs/behavior.py]:
```suggestion
		bpod_filepath = next(bpod_dir.glob('*.mat'))
		trial_info = load_bpod_matfile(key, bpod_filepath )
```

Comment (ttngu207) [reply adamacs/behavior.py]: Let's confirm if checking by just an extension of `.mat` is sufficient to identify the bpod mat file. Could there be other `.mat` files in the directory?

File: adamacs/behavior.py [hunk 3]
```diff
@@ -113,8 +93,50 @@ class TrialVariable(dj.Part):
...
+		schema.Trial.insert(trial_info,ignore_extra_fields=True)
+		schema.Event.insert(trial_info,ignore_extra_fields=True)
```
Comment (ttngu207) [inline adamacs/behavior.py]:
```suggestion
		Trial.insert(trial_info, ignore_extra_fields=True)
		Event.insert(trial_info, ignore_extra_fields=True)
```

File: adamacs/behavior.py [hunk 4]
```diff
@@ -113,8 +93,50 @@ class TrialVariable(dj.Part):
...
+	"""
+	Loading routine for behavioral file, bpod .mat
+	"""
+	#Loading the file
+	SessionData = spio.loadmat(matlab_filepath.as_posix(),
+                            squeeze_me=True, struct_as_record=False)['SessionData']
+
+	# Add to dict for insertion
+  for trial in range(SessionData.nTrials):
+  	trial_info['start_time'] = SessionData.RawData.OriginalEventTimestamps[trial]
```
Comment (ttngu207) [inline adamacs/behavior.py:129]: This block doesn't quite make sense to me, I think you meant for this to be place holder, to demonstrate the steps? If so let's comment this out

**Review by ttngu207 — CHANGES_REQUESTED** [review]
Looks good overall, I'd suggest moving the entire `BehaviorIngest` and the related ingestion functions to a different package (similar to MAP structure).

The `BehaviorIngest` is a ingestion-related table to facilitate the ingestion routine, and should not be a part of the main `behavior` pipeline - i.e. downstream users do not need to see or even be aware of this table.

File: adamacs/behavior.py (hunk 2 above)
Comment (CBroz1) [reply adamacs/behavior.py]: Certainly. Digging through the shared example data, there were other filetypes in the same directory (.m, .fig), but all `.mat` were recordings. I'll confirm this explicitly at our meeting.

File: adamacs/behavior.py (hunk 4 above)
Comment (CBroz1) [reply adamacs/behavior.py:129]: Sure, I'll comment this out and move both the table and the function skeleton to an embedded module.

**Pull request merged**